automerge-frontend/src/state_tree/mod.rs

```diff
@@ -1019,24 +1019,27 @@ struct CursorState {
 }
 
 #[derive(Debug, PartialEq, Clone)]
-struct Cursors(im_rc::HashMap<amp::ObjectId, Vec<CursorState>>);
+struct Cursors(HashMap<amp::ObjectId, Vec<CursorState>>);
 
 impl Cursors {
     fn new() -> Cursors {
-        Cursors(im_rc::HashMap::new())
+        Cursors(HashMap::new())
     }
 
     fn new_from(cursor: CursorState) -> Cursors {
-        Cursors(im_rc::hashmap! {
+        Cursors(maplit::hashmap! {
             cursor.referred_object_id.clone() => vec![cursor],
         })
     }
 
-    fn union(&self, other: Cursors) -> Cursors {
-        Cursors(self.0.clone().union_with(other.0, |mut c1, c2| {
-            c1.extend(c2);
-            c1
-        }))
+    fn union(&mut self, other: Cursors) {
```

Comment:
Given that this is no longer an immutable operation should we rename it to something like `extend`?

Reply:
Yeah that makes sense, done.